qiskit/visualization/matplotlib.py

```diff
@@ -759,8 +759,12 @@ def _draw_ops(self, verbose=False):
                     self._condition(node, cond_xy)
 
                 # draw measure
-                if isinstance(op, Measure):
-                    self._measure(node)
+                if isinstance(op, (Measure, MeasurePauli)):
+                    if isinstance(op, Measure):
+                        basis = "Z"
+                    else:
+                        basis = op.params
+                    self._measure(node, basis=basis)
```

Comment:
This is an example of a bug mentioned in https://github.com/Qiskit/qiskit-terra/pull/5311/files#r810538319

A possible work around for this case might look something like this (is did not test it tho)

```suggestion
                if isinstance(op, MeasurePauli):
                    self._measure(node, basis=op.params[0])
                elif isinstance(op, Measure):
                    self._measure(node, basis='Z')
```